src/OSPSuite.Infrastructure.Import/Core/IDataSet.cs

```diff
@@ -34,10 +35,9 @@ public void AddData(IEnumerable<ParsedDataSet> range)
          _data.AddRange(range);
       }
 
-      public void ThrowsOnNan(double indicator)
+      public bool NanValuesExist(double indicator)
       {
-         if (_data.Any(dataSet => dataSet.Data.Any(pair => pair.Key.ColumnInfo.IsMandatory && pair.Value.Any(point => point.Measurement == indicator || double.IsNaN(point.Measurement)))))
-            throw new NanException();
+         return _data.Any(dataSet => dataSet.Data.Any(pair => pair.Key.ColumnInfo.IsMandatory && pair.Value.Any(point => point.Measurement == indicator || double.IsNaN(point.Measurement))));
```

Comment:
can we comment a bit on this line ...please. And maybe we can break the logic a bit here because it's very complicated to decipher
Also run resharper on this file